src/lib.rs

```diff
@@ -621,6 +636,68 @@ mod test {
         quickcheck::quickcheck(wal as fn(u8) -> TestResult);
     }
 
+    #[test]
+    fn check_last_index() {
+        init_logger();
+        fn check(entry_count: u8) -> TestResult {
+            let dir = tempdir::TempDir::new("wal").unwrap();
+            let mut wal = Wal::with_options(
+                &dir.path(),
+                &WalOptions {
+                    segment_capacity: 80,
+                    segment_queue_len: 3,
+                },
+            )
+            .unwrap();
+            let entries = EntryGenerator::new()
+                .into_iter()
+                .take(entry_count as usize)
+                .collect::<Vec<_>>();
+
+            for entry in &entries {
+                wal.append(entry).unwrap();
+            }
+            let last_index = wal.last_index();
+            if wal.entry(last_index).is_none() && wal.num_entries() != 0 {
```

Comment:
Why do we need `&& wal.num_entries() != 0` ?
Is it because `entry_count` can be zero via Quickcheck?

Reply:
Yes, and in that case entry will be none

Reply:
nitpick: If there is no value in testing `entry` equals to zero, maybe we could return early with `return TestResult::discard();` in that case

Reply:
We still want to test that `last_index` will not panic

Reply:
It actually did when there was no zero check and it overflowed

Reply:
let's keep it then!